sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -706,6 +714,14 @@ class Dataset[T] private[sql](
    * plan may grow exponentially. Local checkpoints are written to executor storage and despite
    * potentially faster they are unreliable and may compromise job completion.
    *
+   * @param eager Whether to checkpoint this dataframe immediately
+   *
+   * @note When checkpoint is used with eager = false, the final data that is checkpointed after
```

Comment:
I think scala client need this doc as well, @zhengruifeng can you followup on this?

Reply:
no problem

Reply:
oh, Spark Connect doesn't support checkpoint

https://github.com/apache/spark/blob/8132e1700c81eff2e814756291b4bf853e764e9a/connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala#L3333-L3347

Reply:
not supported in Python Client either:
https://github.com/apache/spark/blob/f4e41e0e318ea1269de5991f4635637e6e5233f3/python/pyspark/sql/connect/dataframe.py#L1691-L1698

Reply:
ok, we are fine then :)